Assets/Mirror/Editor/Weaver/Readers.cs

```diff
@@ -152,16 +152,16 @@ static MethodDefinition GenerateArrayReadFunc(TypeReference variable, int recurs
             ILProcessor worker = readerFunc.Body.GetILProcessor();
 
             worker.Append(worker.Create(OpCodes.Ldarg_0));
-            worker.Append(worker.Create(OpCodes.Call, Weaver.NetworkReadUInt16));
+            worker.Append(worker.Create(OpCodes.Call, Weaver.NetworkReaderReadPackedInt32));
             worker.Append(worker.Create(OpCodes.Stloc_0));
-            worker.Append(worker.Create(OpCodes.Ldloc_0));
 
+            worker.Append(worker.Create(OpCodes.Ldloc_0));
+            worker.Append(worker.Create(OpCodes.Ldc_I4_0));
             Instruction labelEmptyArray = worker.Create(OpCodes.Nop);
-            worker.Append(worker.Create(OpCodes.Brtrue, labelEmptyArray));
+            worker.Append(worker.Create(OpCodes.Bge, labelEmptyArray));
 
-            // return empty array
-            worker.Append(worker.Create(OpCodes.Ldc_I4_0));
-            worker.Append(worker.Create(OpCodes.Newarr, variable.GetElementType()));
+            // return null
```

Comment:
maybe add a comment like the '//return null' here to the above worker code too?

Reply:
I added some comments,  but I did not understand what you were asking